src/Microsoft.AspNetCore.Server.Kestrel.Core/Internal/Http/Frame.FeatureCollection.cs

```diff
@@ -229,10 +229,32 @@ string IHttpRequestIdentifierFeature.TraceIdentifier
             }
         }
 
-        MinimumDataRate IHttpRequestBodyMinimumDataRateFeature.MinimumDataRate
+        double IHttpMinRequestBodyDataRateFeature.BytesPerSecond
         {
-            get => RequestBodyMinimumDataRate;
-            set => RequestBodyMinimumDataRate = value;
+            get => MinRequestBodyDataRate.BytesPerSecond;
+            set
+            {
+                if (value < 0)
+                {
+                    throw new ArgumentOutOfRangeException(nameof(value), CoreStrings.NonNegativeNumberRequired);
+                }
+
+                MinRequestBodyDataRate.BytesPerSecond = value;
+            }
+        }
+
+        TimeSpan IHttpMinRequestBodyDataRateFeature.GracePeriod
+        {
+            get => MinRequestBodyDataRate.GracePeriod;
+            set
+            {
+                if (value < TimeSpan.Zero)
```

Comment:
Is there a way to share the validation logic between this and the MinimumDataRate class?